web/app.js

```diff
@@ -1512,6 +1513,36 @@ if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
   };
 }
 
+function loadFakeWorker() {
+  return new Promise(function(resolve, reject) {
+    if (typeof PDFJSDev === 'undefined' || !PDFJSDev.test('PRODUCTION')) {
+      if (typeof SystemJS === 'object') {
+        SystemJS.import('pdfjs/core/worker').then((worker) => {
+          window.pdfjsNonProductionPdfWorker = worker;
+          resolve();
+        });
+      } else if (typeof require === 'function') {
+        window.pdfjsNonProductionPdfWorker = require('../src/core/worker.js');
+        resolve();
+      } else {
+        reject(new Error(
+          'SystemJS or CommonJS must be used to load fake worker.'));
+      }
+    } else {
+      let script = document.createElement('script');
+      script.src = PDFJS.workerSrc;
+      script.onload = function() {
+        resolve();
+      };
+      script.onerror = function() {
+        reject(new Error(`Cannot load fake worker at: ${script.src}`));
+      };
+      (document.getElementsByTagName('head')[0] || document.body).
```

Comment:
This can be simplified to `document.head || document.body`. Only use `document.head` is only undefined in IE8-, and supported by all other browsers: https://developer.mozilla.org/en-US/docs/Web/API/Document/head

I usually do `document.head || document.documentElement` (since `<head>` should exist; if it does not, then `document.body` may also be empty, but the root element should always exist, so that's a safe fallback).

Reply:
I was just copying the existing pattern used in https://github.com/mozilla/pdf.js/blob/master/web/app.js#L1515-L1534; should we simply change both them?

Reply:
You can do that in a separate commit (to keep the refactoring separate from the actual feature).

Reply:
> I usually do `document.head || document.documentElement`

This suggesting has now been implemented (only in the newly added code); thank you!

Reply:
> I usually do `document.head || document.documentElement`

This suggestion has now been implemented (only in the newly added code); thank you!